bambi/plots/effects.py

```diff
@@ -41,6 +44,260 @@ def __post_init__(self):
         self.upper_bound_name = f"upper_{self.upper_bound * 100}%"
 
 
+@dataclass
+class Estimate:
+    mean: Dict[str, xr.DataArray]
+    bounds: Dict[str, xr.Dataset]
+    use_hdi: bool
+    bounds_list: list = field(init=False)
+    lower: xr.DataArray = field(init=False)
+    higher: xr.DataArray = field(init=False)
+
+    def __post_init__(self):
+        """
+        Parses the mean and bounds dictionaries into arrays for inserting
+        the 'mean', 'lower', and 'upper' columns into the summary dataframe.
+        """
+        self.bounds_list = [self.bounds[key] for key in self.bounds]
+        self.mean = np.array(list(self.mean.values())).flatten()
+
+        if self.use_hdi:
+            data_var = list(self.bounds_list[0].data_vars)[0]
+            self.lower = np.array(
+                [self.bounds[key][data_var].sel(hdi="lower") for key in self.bounds]
+            ).flatten()
+            self.higher = np.array(
+                [self.bounds[key][data_var].sel(hdi="higher") for key in self.bounds]
+            ).flatten()
+        else:
+            lower = self.bounds_list[0].coords["quantile"].values[0]
+            higher = self.bounds_list[0].coords["quantile"].values[1]
+            self.lower = np.array(
+                [self.bounds[key].sel(quantile=lower) for key in self.bounds]
+            ).flatten()
+            self.higher = np.array(
+                [self.bounds[key].sel(quantile=higher) for key in self.bounds]
+            ).flatten()
+
+
+SUPPORTED_SLOPES = ("dydx", "eyex")
+SUPPORTED_COMPARISONS = {
+    "diff": lambda x, y: x - y,
+    "ratio": lambda x, y: x / y,
+}
```

Comment:
Constants should be in the beginning of the code (after imports, before defining dataclasses, functions, etc.)